File: Q1_Cpp/src/Shipment/Stock/Stock.cpp
#include "../../../include/Shipment/Stock/Stock.h"

// Constructor for Stock
Stock::Stock(std::shared_ptr<Product> product, int quantity)
    : product(product), quantity(quantity) {
}

// Getter and setter for product
std::shared_ptr<Product> Stock::getProduct() const {
    return product;
}

void Stock::setProduct(std::shared_ptr<Product> product) {
    this->product = product;
}

// Getter and setter for quantity
int Stock::getQuantity() const {
    return quantity;
}

void Stock::setQuantity(int quantity) {
    this->quantity = quantity;
}

File: Q4/src/main/java/com/yourorganization/Shipment/Stock/Stock.java
package com.yourorganization.Shipment.Stock; // Package for shipment and stock management

import com.yourorganization.Product.Product;

public class Stock {
    private Product product; // The product itself
    private int quantity; // Quantity of the product in stock

    // Constructor for StockItem
    public Stock(Product product, int quantity) {
        this.product = product; // Setting the product
        this.quantity = quantity; // Setting the quantity
    }

    // Getter for product
    public Product getProduct() {
        return product;
    }

    // Setter for product
    public void setProduct(Product product) {
        this.product = product;
    }

    // Getter for quantity
    public int getQuantity() {
        return quantity;
    }

    // Setter for quantity
    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }
}
